tools/nni-auto-tune/main.py

```diff
@@ -0,0 +1,230 @@
+    return mean, qps
+
+
+def grid_search(params):
+    param_num = len(params)
+    params = list(params.items())
+    max_param_choices = max([len(p[1]) for p in params])
+    temp = []
+    for i in range(max_param_choices):
+        temp += [i for _ in range(param_num)]
+    for c in set(itertools.permutations(temp, param_num)):
+        res = {}
+        for i in range(len(c)):
+            if c[i] >= len(params[i][1]):
+                break
+            else:
+                res[params[i][0]] = params[i][1][c[i]]
+        if len(res) == param_num:
+            yield res
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument(
+        '--train_file',
+        help='the data file to load training points from, '
+        'could be text file, binary flie or ann-benchmark format hdf5 file',
+        default='glove-100-angular.hdf5')
+    parser.add_argument(
+        '--query_file',
+        help='the data file to load query points from, if you use '
+        'ann-benchmark format hdf5 file in train_file, this should be None',
+        default=None)
+    parser.add_argument(
+        '--label_file',
+        help=
+        'the data file to load groundtruth index from, only support text file',
+        default=None)
+    parser.add_argument('--algorithm',
+                        help='the name of SPTAG algorithm',
+                        default="BKT")
+    parser.add_argument("--k",
+                        default=10,
+                        type=int,
+                        help="the number of near neighbours to search for")
+    parser.add_argument("--distance",
+                        default='angular',
+                        help="the type of distance for searching")
+    parser.add_argument("--dim",
+                        default=100,
+                        type=int,
+                        help="the dimention of training vectors")
+    parser.add_argument("--data_type",
+                        default="float32",
+                        help="the data type of input vectors")
+    args = parser.parse_args()
+
+    if args.train_file.endswith(".hdf5"):
+        # ann-benchmark format hdf5 file got all we want, so args like distance are ignored
+        data_reader = HDF5Reader(args.train_file)
+        X_train, X_test = data_reader.readallbatches()
+        distance = data_reader.distance
+        dimension = data_reader.featuredim
+        label = data_reader.label
+    else:
+        X_train = DataReader(args.train_file,
+                             args.dim,
+                             batchsize=-1,
+                             datatype=args.data_type).readbatch()[1]
+        X_test = DataReader(args.query_file,
+                            args.dim,
+                            batchsize=-1,
+                            datatype=args.data_type).readbatch()[1]
```

Comment:
`X_train`, `X_test` might be type float64, which is not supported by SPTAG.
Please add something like this:
```
if X_train.dtype == '<f8' or X_test.dtype == '<f8':
    X_train = X_train.astype('float32')
    X_test = X_test.astype('float32')
```